### PR TITLE
Seeding last fixes (Nuke)

### PR DESCRIPTION
**Pull request by Kasterov**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: Streetcode/DbUpdate/Program.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
+﻿using DbUp;
+using Microsoft.Extensions.Configuration;
+
+public class Program
+{
+    static int Main(string[] args)
+    {
+        string seedingPath = Directory.GetCurrentDirectory() + @"\Streetcode.DAL\Persistence\ScriptsMigration\";
```
Comment (maxym-lanchevych) [inline Streetcode/DbUpdate/Program.cs:8]: It's better to use Path.Combine for this purpose

File: Streetcode/DbUpdate/Program.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        string migrationPath = Directory.GetCurrentDirectory() + @"\Streetcode.DAL\Persistence\ScriptsSeeding\";
+
+        var environment = Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT") ?? "Local";
+
+        var configuration = new ConfigurationBuilder()
+            .SetBasePath(Directory.GetCurrentDirectory() + @"\Streetcode.WebApi")
```
Comment (maxym-lanchevych) [inline Streetcode/DbUpdate/Program.cs:14]: Path.Combine

File: Streetcode/DbUpdate/Program.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+        var connectionString = configuration.GetConnectionString("DefaultConnection");
+
+        string pathToScript = "";
+
+        while (pathToScript != "1" && pathToScript != "2")
+        {
+            Console.WriteLine("Enter '1' to MIGRATE or '2' to SEED db:");
+            pathToScript = Console.ReadLine();
+        }
```
Comment (maxym-lanchevych) [inline Streetcode/DbUpdate/Program.cs:28]: let's move this to args, so if we pass something like "-s or -seeding" we will seed database, if "-m or -migrate" we will migrate only. If this is not provided let's set default to migrate

File: Streetcode/Streetcode.WebApi/Extensions/WebApplicationExtensions.cs [hunk 4]
```diff
@@ -1,92 +1,92 @@
...
 
-                var scriptFiles = Directory.GetFiles($"{projRootDirectory}/{scriptsFolderPath}");
+                    var scriptFiles = Directory.GetFiles($"{projRootDirectory}/{scriptsFolderPath}");
 
-                var filesContexts = await Task.WhenAll(scriptFiles.Select(file => File.ReadAllTextAsync(file)));
-                transaction = streetcodeContext.Database.BeginTransaction();
-                foreach (var task in filesContexts)
+                    var filesContexts = await Task.WhenAll(scriptFiles.Select(file => File.ReadAllTextAsync(file)));
+                    transaction = streetcodeContext.Database.BeginTransaction();
+                    foreach (var task in filesContexts)
```
Comment (maxym-lanchevych) [inline Streetcode/Streetcode.WebApi/Extensions/WebApplicationExtensions.cs:29]: it's not a task

File: Streetcode/Streetcode.WebApi/Extensions/WebApplicationExtensions.cs [hunk 5]
```diff
@@ -1,92 +1,92 @@
...
             {
-                var projRootDirectory = Directory.GetParent(Environment.CurrentDirectory)?.FullName!;
+                var streetcodeContext = scope.ServiceProvider.GetRequiredService<StreetcodeDbContext>();
+
+                IDbContextTransaction transaction = null;
+                try
+                {
+                    var projRootDirectory = Directory.GetParent(Environment.CurrentDirectory)?.FullName!;
+
+                    var scriptFiles = Directory.GetFiles($"{projRootDirectory}/{scriptsFolderPath}");
```
Comment (maxym-lanchevych) [inline Streetcode/Streetcode.WebApi/Extensions/WebApplicationExtensions.cs:66]: Path.Combine

File: build/Targets/DatabaseBuild.cs [hunk 6]
```diff
@@ -40,29 +44,48 @@ partial class Build
...
+            if (queryType == "SEED")
+            {
+                queryType = "ScriptsSeeding";
+            } 
+            else 
+            {
+                queryType = "ScriptsMigration";
+            }
+
+            var dbPath = Directory.GetCurrentDirectory() + @"\Streetcode.WebApi";
```
Comment (maxym-lanchevych) [inline build/Targets/DatabaseBuild.cs:73]: Path.Combine

File: build/Targets/DatabaseBuild.cs [hunk 7]
```diff
@@ -40,29 +44,48 @@ partial class Build
...
+            var outputScriptPath = Directory.GetCurrentDirectory() + @$"\Streetcode.DAL\Persistence\{queryType}\";
+
+            PowerShell(@$"dotnet ef migrations script --idempotent --output {outputScriptPath}{queryName}.sql  --project {dbPath}");
```
Comment (maxym-lanchevych) [inline build/Targets/DatabaseBuild.cs:76]: Let's add here dependent step that will install ef globally, as right now if you don't have it this command will fail
The same dependent step let's add to CreateDatabase target

File: Streetcode/DbUpdate/Program.cs (hunk 3 above)
Comment (Kasterov) [reply Streetcode/DbUpdate/Program.cs:28]: got it!

File: Streetcode/DbUpdate/Program.cs (hunk 1 above)
Comment (Kasterov) [reply Streetcode/DbUpdate/Program.cs:8]: got it!

**Review by Tysyatsky — APPROVED** [review]

**Pull request merged**